plutus-ledger-api/src/Plutus/V1/Ledger/Address.hs

```diff
@@ -11,50 +11,77 @@
 module Plutus.V1.Ledger.Address (
     Address (..),
     pubKeyAddress,
+    pubKeyHashAddress,
     scriptAddress,
     scriptHashAddress,
+    toPubKeyHash,
+    toValidatorHash,
+    stakingCredential
     ) where
 
-import           Codec.Serialise.Class     (Serialise)
-import           Control.DeepSeq           (NFData)
-import           Data.Aeson                (FromJSON, FromJSONKey (..), ToJSON, ToJSONKey (..))
-import           Data.Hashable             (Hashable)
+import           Codec.Serialise.Class       (Serialise)
+import           Control.DeepSeq             (NFData)
+import           Data.Aeson                  (FromJSON, FromJSONKey (..), ToJSON, ToJSONKey (..))
+import           Data.Hashable               (Hashable)
 import           Data.Text.Prettyprint.Doc
-import           GHC.Generics              (Generic)
-import qualified PlutusTx                  as PlutusTx
-import qualified PlutusTx.Eq               as PlutusTx
+import           GHC.Generics                (Generic)
+import qualified PlutusTx                    as PlutusTx
+import qualified PlutusTx.Bool               as PlutusTx
+import qualified PlutusTx.Eq                 as PlutusTx
 
+import           Plutus.V1.Ledger.Credential (Credential (..), StakingCredential)
 import           Plutus.V1.Ledger.Crypto
-import           Plutus.V1.Ledger.Orphans  ()
+import           Plutus.V1.Ledger.Orphans    ()
 import           Plutus.V1.Ledger.Scripts
 
--- | A payment address using a hash as the id.
-data Address = PubKeyAddress PubKeyHash
-    | ScriptAddress ValidatorHash
+-- | Address with two kinds of credentials, normal and staking
+data Address = Address{ addressCredential :: !Credential, addressStakingCredential :: !(Maybe StakingCredential)}
```

Comment:
let's not include the bangs, I don't think they do anything and it'll make people think that they do

Reply:
Good point. I'll remove them.